vite.config.js

```diff
@@ -0,0 +1,29 @@
+import {defineConfig, loadEnv } from 'vite'
+
+// Match ports in .ddev/config.yaml -> web_extra_exposed_ports
+const HTTP_PORT = 3000;
+const HTTPS_PORT = 3001;
+
+export default defineConfig(({command, mode}) => {
+    const env = loadEnv(mode, process.cwd(), '');
+    const originPort = env.PRIMARY_SITE_URL.startsWith('https') ? HTTPS_PORT : HTTP_PORT;
+
+    return {
+        base: command === 'serve' ? '' : '/dist/',
+        build: {
+            manifest: true,
+            outDir: './web/dist/',
+            rollupOptions: {
+                input: {
+                    app: 'src/js/app.js',
+                }
+            },
+        },
+        server: {
+            host: '0.0.0.0',
+            strictPort: true,
+            port: HTTP_PORT,
+            origin: env.PRIMARY_SITE_URL + ':' + originPort,
```

Comment:
This line is also needed to get webfonts, images, etc to work properly in dev mode.

You have to tell Vite the precise URL to use for any files referenced in your CSS & JS files.